### ISSUE TITLE
Is lower validation loss in smaller steps, means better generate result in bigger steps?

### ISSUE THREAD

**Issue by JohnHerry** [bfs18/e2_tts#17]
I see in the reference.py,  the final ode runs N=100 steps for diffusion. but during training, the validation take N=20 steps for diffusion. and,  according to the project config, it will preserve 3 checkpoints with the best validation performance.  Is that means when N=20 steps get better denoise result, it will surely get better result when  N =100 steps？

**Comment by bfs18** [comment]
Hi @JohnHerry Generally speaking, 10 steps of sampling is usually sufficient. And models that perform well with 20 steps typically also perform well with 100 steps.

**Comment by JohnHerry** [comment]
> Hi @JohnHerry Generally speaking, 10 steps of sampling is usually sufficient. And models that perform well with 20 steps typically also perform well with 100 steps.

Thank you.    
 My training instance has run over 500k steps, and during my test, 20 steps is enough for good speech, and for 50, or 100 steps, may generate more clean speech but some times worse prosody, but 10 steps generate worse audibility.    
and during my test, E2_TTS will also have some hallucination.  the generated speech may contain nonsense,  repeating,  skipping words just like in AR SLM  models.   and it seems that the Prompt speech just provide the speaker timbre to the generated speech,  The speech prosody generated just like something randomly sampled from dataset distribution,  but not following the Prompt speech.    
and as to the Duration Predictor,  In my tests, it is prone to predict larger duration then actually needed. and as a result , the generated speech takes more random  and undesired prosodys. example. a ground truth speech may only 7 seconds , but the duration predictor force the E2_TTS to infill mel to the length of 11 seconds, as a result it may produce some strange breaks[silence segment] in the speech.
